extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -362,8 +364,9 @@ export async function newDeployment(
         const python = await getPythonInterpreterPath();
         const inspectResponse = await api.configurations.inspect(
           {
-            dir: ".",
-            recursive: true,
+            dir: projectDir ? projectDir : ".",
+            entrypoint: entryPoint,
+            recursive: projectDir ? false : true,
```

Comment:
when we pass in a projectDir, we always disable recursive behavior. This was the way the logic / use worked out. In the future, if we do want something different, obviously we could add a new param to be passed in.

Reply:
This was incorrect. We collect recursively, but we also store them by subdirectory, so we can find unique names. I've added the recursive back in.